docs/mindsdb-handlers.mdx

```diff
@@ -0,0 +1,75 @@
+│  ├─ install.py                           # Script that installs all handler dependencies
+```
+
+## Structure of a Handler
+
+In technical terms, a handler is a self-contained Python package having everything required for MindsDB to interact with it. It includes aspects like dependencies, unit tests, and continuous integration logic. It is up to the author to determine the nature of the package, for example, closed or open source, version control, and more. Although, we encourage opening pull requests to expand the default set of supported tools.
+
+The entry point for a database handler is a class definition that should inherit directly from the [mindsdb.integrations.libs.base.DatabaseHandler](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L102) class and thus indirectly from the [mindsdb.integrations.libs.base.BaseHandler](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L8) class, which defines all the methods that must be overwritten in order to achieve a functional implementation.
+
+And the entry point for an ML handler is a class definition that should inherit from the [mindsdb.integrations.libs.base.BaseMLEngine](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L122) class, which defines all the methods that must be overwritten in order to achieve a functional implementation.
+
+<Note>
+    The handler's structure is not enforced, and the package design is up to the author.
+</Note>
+
+## Things to Remember
+
+Here are a few points to keep in mind while implementing a handler.
+
+### Handlers Inherit from Base Classes
+
+Inherit from the [`DatabaseHandler`](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L102) class when adding a new database handler. For more info, visit our [doc page here](/contribute/integrations/).
+
+Inherit from the [`BaseMLEngine`](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L122) class when adding a new ML handler. For more info, visit our [doc page here](/contribute/ml-handlers/).
+
+### Parsing SQL
+
+Whenever you want to parse a string that contains SQL, we strongly recommend using the `mindsdb_sql` package. It provides a parser that fully supports the MindsDB SQL dialect and partially the standard SQL dialect. There is also a *render* feature to map other dialects into the already supported ones.
+
+### Formatting Output
+
+When it comes to building the response of the public methods, the output should be wrapped by the [mindsdb.integrations.libs.response.HandlerResponse](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/response.py#L7) or [mindsdb.integrations.libs.response.HandlerStatusResponse](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/response.py#L32) class. These classes are used by the MindsDB executioner to orchestrate and coordinate multiple handler instances in parallel.
```

Comment:
Let's specify that this applies to the data handlers. For ML handlers, this is automatically done by an intermediate wrapper (`BaseMLEngineExec`) so the contributor wouldn't need to worry about it.